docs/automated_testing.md

```diff
@@ -0,0 +1,509 @@
+
+Secrets
+
+PLAYGROUND_PASSWORD
+PLAYGROUND_ID
+PLAYGROUND_EMAIL
+-->
+
+## Summary
+
+1. You write and edit `.feature` test files in your Sources folder.
+1. There's a maximum of 2 minutes per individual Step or table row. Longer than that and the test will error. That can't be customized yet.
+1. Tests are run when you commit.
+1. Tests can download PDF files, but humans have to review them to see if they're right.
+1. Tests that error should create screenshots from when the error happened.
+1. Tests create report files. They don't have a lot right now, but they might have some clues when something unexpected happens.
+
+For interacting with things on GitHub, look for how to use the [Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions#viewing-the-jobs-activity) tab.
+
+Give us feedback and ideas by making issues at https://github.com/plocket/docassemble-cucumber/issues.
+
+## Example
+
+The tests use the [gherkin language and syntax](https://cucumber.io/docs/gherkin/reference/). Here's a complex example for quick reference of some of our features:
+
+```
+@children
+Feature: The user has children
+
+  @fast @child_benefits
+  Scenario: child has benefits
+    Given I start the interview at "some_file_name"
+    And I get to the question id "benefits" with this data:
+      | var | value | trigger |
+      | x[i].name.first | Cashmere | children[0].name.first |
+      | x[i].name.first | Casey | children[1].name.first |
+      | x.there_are_any | True | children.there_are_any |
+      | x.there_is_another | 2 | children.there_is_another |
+    When I set the var "benefits['SSI']" to "true"
+    And I tap to continue
+    Then the question id should be "download"
+    And I download "some_motion.pdf"
+
+```
+
+## First test
+
+You can write a really simple test right away that just makes sure your a file runs using the name of the file. Write a `Scenario` for each file you want to test.
+
+```
+Feature: Interviews load
+
+  Scenario: The 209A loads
+    Given I start the interview at "ma_209a_package"
+
+  Scenario: The Plaintiff's Motion to Modify loads
+    Given I start the interview at "plaintiffs_motion_to_modify_209a"
+```
+
+More complex tests might wait till your code is pretty much how you want it. Every time you change your variable names, you may have to update the tests.
+
+## Story tables syntax
+
+Very basically, a **story table** contains a list of variables and values. The order in which you list the rows does not matter, which gives a lot more flexibility when developers are changing around the order of questions. Whenever pages with those variables appear, they will be given the value listed.
+
+Tip: it can be easy to lose track of what variables you've included, so it can be useful keep the rows in alphabetical order.
```

Comment:
```suggestion
:::tip 
It can be easy to lose track of what variables you've included, so it can be useful keep the rows in alphabetical order.
:::
```